sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/ConfigurationClientCredentials.java

```diff
@@ -82,23 +82,37 @@ public String getBaseUri() {
      */
     Mono<Map<String, String>> getAuthorizationHeadersAsync(URL url, String httpMethod, Flux<ByteBuffer> contents) {
```

Comment:
do we need async version ? I don't see any io here.
The caller could just call sync version (wrapped in some reactore operator)